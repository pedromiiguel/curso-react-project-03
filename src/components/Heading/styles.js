import styled, { css } from 'styled-components';

export const Title = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.primaryColor};
  `}
`;
