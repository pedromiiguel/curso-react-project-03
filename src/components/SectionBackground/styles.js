import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${background ? theme.colors.primaryColor : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.primaryColor};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
