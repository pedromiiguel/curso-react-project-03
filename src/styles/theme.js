export const theme = {
  colors: {
    primaryColor: '#0a1128',
    secondaryColor: '#DC143C',
    lightBg: '#ffffff',
    darkBg: '#0A1128',
  },

  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
};
