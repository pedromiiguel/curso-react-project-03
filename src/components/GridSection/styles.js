/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    /* grid-template-columns: repeat(auto-fill, minmax(280px, 1fr)); */
    grid-template-columns: 1fr 1fr 1fr;

    gap: ${theme.spacings.large};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      width: 80%;
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -2rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
