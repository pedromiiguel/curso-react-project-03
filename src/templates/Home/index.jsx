/* eslint-disable no-unused-vars */
import { useEffect, useState } from 'react';
import { Base } from '../Base';
import { mockBase } from '../Base/mock';
import { Loading } from '../Loading';
import { PageNotFound } from '../PageNotFound';
import { GridTwoColumns } from '../../components/GridTwoColumns';
import { GridContent } from '../../components/GridContent';
import { GridSection } from '../../components/GridSection';
import { GridImage } from '../../components/GridImage';
import { mapData } from '../../api/map-data';

import config from '../../config';

function Home() {
  const [data, setData] = useState([]);

  useEffect(() => {
    const load = async () => {
      try {
        const data = await fetch(config.url);
        const json = await data.json();
        const pageData = mapData(json);
        setData(pageData[0]);
      } catch (e) {
        setData(undefined);
      }
    };

    load();
  }, []);

  useEffect(() => {
    if (data === undefined) {
      return (document.title = 'Página não encotrada');
    }

    if (data && !data.slug) {
      return (document.title = 'Carregando ...');
    }

    if (data && data.title) {
      return (document.title = data.title);
    }
  }, [data]);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml, slug } = data;
  const { links, text, link, srcImg } = menu;

  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      logoData={{ text, link, srcImg }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${slug}-${index}`;

        if (component === 'section.section-two-columns') {
          return <GridTwoColumns key={key} {...section} />;
        }

        if (component === 'section.section-content') {
          return <GridContent key={key} {...section} />;
        }

        if (component === 'section.section-grid-text') {
          return <GridSection key={key} {...section} />;
        }

        if (component === 'section.section-grid-image') {
          return <GridImage key={key} {...section} />;
        }
      })}
    </Base>
  );
}

export default Home;
