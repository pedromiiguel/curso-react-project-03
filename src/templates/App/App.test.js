import { screen } from '@testing-library/react';
import Home from '.';
import { renderTheme } from '../../styles/render-theme';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);
  debug();
  const headingContainer = screen.getByRole('heading', { name: 'Hello World!' }).parentElement;

  expect(headingContainer).toHaveStyle({
    background: 'pink',
  });

  expect(headingContainer).toMatchSnapshot();

  expect(headingContainer).toHaveStyleRule('background', 'pink');
});
