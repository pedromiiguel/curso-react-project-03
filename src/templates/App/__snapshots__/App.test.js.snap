// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background: pink;
}

<div
  class="c0"
>
  <h1>
    Hello World!
  </h1>
</div>
`;
